### DISCUSSION TITLE
Q: `createServerFn` doesn't cope with Branded Types well.

### DISCUSSION THREAD

**Discussion by craftgear** [TanStack/router#2907] [Q&A]
How can I avoid this error when I use Branded Types?

Here is a minimal reproduce-able code :
```typescript
import { z } from "zod";

const FooSchema = z.object({
  id: z.string().brand("FooId"),
});
type Foo = z.infer<typeof FooSchema>;

const getFoo = createServerFn({
  method: "GET",
}).handler(() => { // here I get a type error.
  return FooSchema.parse({ id: "foo" });
});

```

And the error is: 
`
Argument of type '() => { id: string & z.BRAND<"FooId">; }' is not assignable to parameter of type 'ServerFn<"GET", undefined, undefined, { id: string & BRAND<"FooId">; }>'.  Type '{ id: string & BRAND<"FooId">; }' is not assignable to type '{ id: { readonly [x: number]: string; toString: "Function is not serializable"; charAt: "Function is not serializable"; charCodeAt: "Function is not serializable"; concat: "Function is not serializable"; ... 46 more ...; [BRAND]: { ...; }; }; } | Promise<...>'.
`

Any thoughts?

**Comment by macklinu** [comment]
Yes I'm hoping that https://github.com/TanStack/router/pull/3849 will get merged and zod branded types will be supported from server functions.

**Reply by craftgear** [reply]
Hey, thanks for letting me know this!

It looks promising on this matter 👍
